source/val/type.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-2016 The Khronos Group Inc.
```

Comment:
KhronosGroup.  This was written under contract to Khronos.  I believe it was as "work for hire".